homeassistant/components/omnilogic/common.py

```diff
@@ -0,0 +1,167 @@
+
+            if "systemId" in item:
+                system_id = item["systemId"]
+                current_id = current_id + (item_kind, system_id)
+                data[current_id] = item
+
+            for kind in ALL_ITEM_KINDS:
+                if kind in item:
+                    data = get_item_data(item[kind], kind, current_id, data)
+
+            return data
+
+        parsed_data = get_item_data(data, "Backyard", (), parsed_data)
+
+        return parsed_data
+
+
+class OmniLogicEntity(CoordinatorEntity):
+    """Defines the base OmniLogic entity."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        kind: str,
+        name: str,
+        item_id: tuple,
+        icon: str,
+    ):
+        """Initialize the OmniLogic Entity."""
+        super().__init__(coordinator)
+
+        bow_id = None
+        entity_data = coordinator.data[item_id]
+
+        backyard_id = item_id[:2]
+        if len(item_id) == 6:
+            bow_id = item_id[:4]
+
+        msp_system_id = coordinator.data[backyard_id]["systemId"]
+        entity_friendly_name = f"{coordinator.data[backyard_id]['BackyardName']} "
+        unique_id = f"{msp_system_id}"
+
+        if bow_id is not None:
+            unique_id = f"{unique_id}_{coordinator.data[bow_id]['systemId']}"
+            entity_friendly_name = (
+                f"{entity_friendly_name}{coordinator.data[bow_id]['Name']} "
+            )
+
+        unique_id = f"{unique_id}_{coordinator.data[item_id]['systemId']}_{kind}"
+
+        if entity_data.get("Name") is not None:
+            entity_friendly_name = f"{entity_friendly_name} {entity_data['Name']}"
+
+        entity_friendly_name = f"{entity_friendly_name} {name}"
+
+        unique_id = unique_id.replace(" ", "_")
+
+        self._kind = kind
+        self._name = entity_friendly_name
+        self._unique_id = unique_id
+        self._item_id = item_id
+        self._icon = icon
+        self._attrs = {}
+        self._msp_system_id = msp_system_id
+        self._backyard_name = coordinator.data[backyard_id]["BackyardName"]
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique, Home Assistant friendly identifier for this entity."""
+        return self._unique_id
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon for the entity."""
+        return self._icon
+
+    @property
+    def device_state_attributes(self):
+        """Return the attributes."""
+        if "Alarms" in self.coordinator.data[self._item_id]:
+            alarms = self.coordinator.data[self._item_id]["Alarms"]
+
+            if len(alarms) != 0:
+                self._attrs["alarm_message"] = alarms[0].get("Message")
+                self._attrs["alarm_comment"] = alarms[0].get("Comment")
+            else:
+                self._attrs["alarm_message"] = None
```

Comment:
We don't add state attribute items at all of the value is None.

Reply:
If the alarms clear I would like the attributes to go blank. With 0.115 and the ability to use attributes as triggers, this will allow us to not build a binary sensor platform just to identify that there is an alarm triggered on the system. Instead an automation can simply look for self._attrs["alarm_message"] change and then act on notification etc including both message and comment knowing that both will be there (even if blank).

Reply:
We still have to follow our standard.

The use case described here tells me we shouldn't add these alarms as general state attributes but make a binary sensor platform for the binary alarms and fire custom events for the other alarms instead.

Reply:
So in that case we would be building a binary sensor alarm sensor for every Hayward entity which supports reporting alarms? 

I supposed the alternative which would eliminate this issue could be setting the message and comments to "Clear" so it is really informative. If a user wants to take advantage of the new features of 115 for alerts it would be possible.

Reply:
We shouldn't use state attributes for the alarm feature. This feature is useful on its own so should be either separate entities or custom events. We only add state attributes for features that require the context of the main state measurement to be useful. Eg light brightness, is not useful without knowing if the light is on or off.

Reply:
OK fair - will remove from here and look at custom events (completely new to me) or include as binary sensor when we get there.